tripy/tripy/frontend/module/pooling.py

```diff
@@ -0,0 +1,138 @@
+#
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from dataclasses import dataclass
+from collections.abc import Sequence
+
+from tripy import export, utils
+from tripy.frontend.module.module import Module
+
+from tripy.common.exception import raise_error
+
+
+@export.public_api(document_under="operations/modules")
+@dataclass
+@utils.constant_fields(["dtype", "normalized_shape"])
+class MaxPool(Module):
```

Comment:
We distinguish modules from functions based on whether the op has weights. So in this case, I think a function would be better.

Reply:
Makes sense, switched to function.